### DISCUSSION TITLE
[A/L] (2021/22) Funzioni - Esercizio 16

### DISCUSSION THREAD

**Discussion by matypist** [sapienzastudentsnetwork/metodi-matematici-per-l-informatica#256] [Esercizi A-L]
![image](https://user-images.githubusercontent.com/69389461/152315824-1bfca85e-dda1-46fc-b4ca-72f3b3dbcd1e.png)

**Comment by LuginiAndrea** [comment]
La prima affermazione è falsa. 
Basta infatti prendere 

- $f: \mathbf{N} \rightarrow \mathbf{N}$ tolto $0$, $f\left(x\right)=x+1$
- $g: \mathbf{N}$ tolto $0$ $\rightarrow \mathbf{N}$ tolto $0$ (valido in quanto per qualsiasi insieme $X$, $X \subseteq X$), $g\left(z\right) = z+1$
Mentre $f$ è suriettiva, non possiamo dire lo stesso di $g$. Infatti preso $\left(w = 1\right) \in \mathbf{N}$ tolto $0$
$$w = g\left(z\right) \Longrightarrow w = z + 1 \Longrightarrow 1 = z + 1 \Longrightarrow z = 0$$
Però $\left(z = 0\right) \notin \mathbf{N}$ tolto $0 \Longrightarrow \nexists z \in Z$ t.c. $1 = g\left(z\right)$
$g$ non è quindi suriettiva.

Anche la seconda affermazione è falsa
Prendiamo

- $f:\mathbf{Z} \rightarrow \mathbf{N}, f\left(x\right) = x^2$
- $g:\mathbf{N} \rightarrow \mathbf{N}, g\left(z\right) = z^2$

Possiamo ora facilmente notare come 
$$\mathbf{Z} - \mathbf{N} = -\mathbf{N} - 0 \Longrightarrow f\left(\mathbf{Z} - \mathbf{N}\right) = f\left(-\mathbf{N} - 0\right) = \left(-\mathbf{N}\right)^2 - f\left(0\right) = \mathbf{N}^2 - 0$$
Ora consideriamo la seconda parte dell'affermazione:
$$f\left(\mathbf{Z}\right) - f\left(\mathbf{N}\right) = \left(f\left(-\mathbf{N}\right) \cup f\left(\mathbf{N}\right)\right) - f\left(\mathbf{N}\right) = \left(\mathbf{N}^2 \cup \mathbf{N}^2\right) - \mathbf{N}^2 = N^2 - N^2 = \emptyset$$
$$\emptyset \neq \mathbf{N}^2 - 0 \Longrightarrow f\left(\mathbf{Z}\right) - f\left(\mathbf{N}\right)  \neq f\left(\mathbf{Z} - \mathbf{N}\right)$$

Non credo nessuno si sorprenderà se dico che anche la terza è falsa.
Prendiamo

- $f: \mathbf{N} \rightarrow \mathbf{N}, f\left(x\right) = x + 1$
- $g: \mathbf{N} \rightarrow \mathbf{N}, g\left(z\right) = z + 1$
E' validissimo definire $g$ così. in quanto $N \subset N$, e non abbiamo nessuna indicazione sulla definizione di $g$.
Notiamo quindi come $g$ è perfettamente equivalente a $f$ per costruzione, quindi 
$$f\left(\mathbf{N}\right) \cup g\left(\mathbf{N}\right) = f\left(\mathbf{N}\right) \cup f\left(\mathbf{N}\right) = f\left(\mathbf{N}\right)$$
Possiamo facilmente vedere come $f$ non è suriettiva, ovvero $\exists y \in \mathbf{N}$ t.c. $\nexists x \in \mathbf{N}$ t.c. $y = f\left(x\right)$. 
Possiamo facilmente controllarlo ponendo $y = 0$.
Quindi $Y \neq f\left(N\right) \Longrightarrow Y \neq f\left(N\right) \cup g\left(\mathbf{N}\right)$
